Dn02/report/report.jmd

```diff
@@ -0,0 +1,258 @@
+# Efficient Computation of the CDF of the Normal Distribution
+**Alen Kurtagić**
+
+# CDF of the Normal Distribution
+The cumulative distribution function (CDF) of a random variable $X$ is defined as
+
+```math
+F_X(x) = P(X \leq x) = \int_{-\infty}^{x} f_X(t) \, dt,
+```
+
+where $f_X(t)$ is the probability density function (PDF) of $X$. The CDF gives the probability that $X$ takes a value less than or equal to $x$.
+
+For the normal distribution $N(\mu, \sigma^2)$, the PDF is
+
+```math
+f_X(x) = \frac{1}{\sqrt{2\pi\sigma^2}} \exp\!\left(-\frac{(x - \mu)^2}{2\sigma^2}\right).
+```
+
+In particular, for the standard normal distribution $N(0,1)$, we have
+
+```math
+f_X(x) = \frac{1}{\sqrt{2\pi}} \exp\!\left(-\frac{x^2}{2}\right).
+```
+
+The corresponding CDF is
+
+```math
+F(x) = \frac{1}{\sqrt{2\pi}} \int_{-\infty}^{x} e^{-t^2/2} \, dt.
+```
+
+
+```julia
+using Plots
+
+using Distributions
+Φ(x) = cdf(Normal(0,1), x)
+
+xs = -3.45:0.1:3.45
+truevals = [Φ(x) for x in xs]
+
+plot(xs, truevals, label="Groundtruth", lw=2, color=:blue)
+xlabel!("x")
+ylabel!("F(x)")
+title!("Standard Normal CDF Groundtruth")
+```
+
+The standard normal distribution is especially important because any normal distribution can be transformed into it. If $X \sim N(\mu, \sigma^2)$, then the standardized variable
+
+```math
+Z = \frac{X - \mu}{\sigma}
+```
+
+follows the standard normal $N(0,1)$. Thus, computing probabilities for *any* normal distribution reduces to evaluating the CDF of $N(0,1)$, which makes it the universal building block for all normal CDF calculations.
+
+# Splitting the Domains
+
+The standard normal CDF has no closed-form expression in elementary functions. Our numerical computation of $F(x)$ must satisfy:
+
+- High precision (relative error ≤ $10^{-10}$).  
+- Uniform computational complexity across the domain.  
+
+Direct quadratures (e.g., Newton–Cotes and Gauss rules) are inefficient, since larger $x$ would require more integration steps. Instead, the domain is split into regions:
+
+- Near 0: Taylor expansion of $\mathrm{erf}(x)$.  
+- Moderate values (2 ≤ x ≤ 5): Gauss–Legendre quadrature (fixed steps).  
+- Large $x > 5$: Asymptotic expansion.  
+
+We can also take advantage of the symmetry of the normal distribution:
```

Comment:
For values smaller than $-5$ this formula would result in large relative error(due to catastrophic subtraction).